store/src/java/com/zimbra/cs/service/mail/Search.java

```diff
@@ -72,6 +73,8 @@
  */
 public class Search extends MailDocumentHandler  {
 
+    public static final String FILE_SHARED = "in:\"Files shared with me\"";
```

Comment:
why do we need this check? client can also send in:20 by using folder id

Reply:
since the folder name was fixed , and no other folder could  have this name or passed in similar format(for subfolder)
hence we had decided to go with this, Let me know your thought on this if requires modification from client can send folder ID , will convey same to Mitesh

Reply:
I think we need to write generic code on server side, client can pass either folder name or folder id in query or any other supported query parameter
we shouldn't hardcode that on server side, is it possible to remove this hardcoded check?
@dasiyogesh any thoughts?

Reply:
@silentsakky we can remove this hardcoded check , but we would require folder ID/folder name in request to do validation in puthits method

Reply:
@silentsakky @dasiyogesh as discussed removed this check and implement check based on folder ID , checking on SOAP API and UI does seem to work